NumberValidator/Validators/IN/vid.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace VidHandling
+{
+    public class InvalidLengthException : Exception { }
+    public class InvalidFormatException : Exception { }
+    public class InvalidChecksumException : Exception { }
+
+    public static class Vid
+    {
+        private static readonly Regex VidRegex = new Regex(@"^[2-9][0-9]{15}$");
+
+        public static string Compact(string number) =>
+            number?.Replace(" ", "").Replace("-", "").Trim() ?? string.Empty;
+
+        public static string Validate(string number)
+        {
+            var compactedNumber = Compact(number);
+
+            if (compactedNumber.Length != 16 || !VidRegex.IsMatch(compactedNumber) || IsPalindrome(compactedNumber))
+            {
+                throw new InvalidFormatException();
+            }
+
+            Verhoeff.Validate(compactedNumber);
+            return compactedNumber;
+        }
+
+        public static bool IsValid(string number)
+        {
+            try
+            {
+                Validate(number);
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        public static string Format(string number)
+        {
+            var compactedNumber = Compact(number);
+            return $"{compactedNumber.Substring(0, 4)} {compactedNumber.Substring(4, 4)} {compactedNumber.Substring(8, 4)} {compactedNumber.Substring(12)}";
+        }
+
+        public static string Mask(string number)
+        {
+            var compactedNumber = Compact(number);
+            return $"XXXX XXXX XXXX {compactedNumber.Substring(12)}";
+        }
+
+        private static bool IsPalindrome(string number) =>
+            number == new string(number.Reverse().ToArray());
+
+        public static string GenerateValidVid()
+        {
+            string vid;
+            do
+            {
+                vid = GenerateRandomVid();
+            } while (IsPalindrome(vid) || !IsValid(vid));
+            return vid;
+        }
+
+        private static string GenerateRandomVid()
+        {
+            var rand = new Random();
```

Comment:
name it properly. what are we saving by typing two letter less?

Reply:
its corrected